packages/types/src/Metadata/v0/Storage.ts

```diff
@@ -69,21 +70,36 @@ export class MapType extends Struct {
 export class PlainType extends Type {
 }
 
-export class StorageFunctionType extends EnumType<PlainType | MapType> {
+export class StorageFunctionType extends EnumType<PlainType | MapType | DoubleMapType> {
```

Comment:
There's no DoubleMap in v0, everything that's related to DoubleMap in this file should be deleted

Reply:
I got ‘FATAL: Unable to initialize the API:  Object.values(...)[index] is not a constructor’ error at EnumType.ts:109:14 without having DoubleMapType here. Added DoubleMapType for fixing this issue.
Here `private static createValue (def: TypesDef, index: number = 0, value?: any): Decoded {
    return {
      index,
      value: new (Object.values(def)[index])(value)
    };
  }
` index is 2 but def only have MapType and PlainType. Call stack says, it's because of `const extrinsics = extrinsicsFromMeta(this.runtimeMetadata.asV0);`

Reply:
Ah, I suspected this would happen...

I think there's a small refactoring to do, have a look at point 1 in https://github.com/polkadot-js/api/issues/858#issue-437173145:
- create v0/ and v4/ folders in type-extrinsic
- move current fromMetadata to fromMetadataV0
- fromMetadata should call fromMetadataV0 if it's v0-3
- in v4/, create a fromMetadataV4, which correctly creates extrinsics from v4

Have a look at type-storage, type-extrinsics should follow the same pattern, and this would allow adding hacks like adding DoubleMap in v0

Reply:
Thank you @amaurymartiny. I will try doing the refactoring following your guide.

Reply:
@AlexWZ0129 I've done similar stuffs, you can reuse code from my refactor branch.

Reply:
@ianhe8x Okay. Thank you, Ian!